File: src/main/java/org/blacksun/graph/NamedGraphNodeFactory.java
package org.blacksun.graph;

import org.jetbrains.annotations.NotNull;

public class NamedGraphNodeFactory implements GraphNodeFactory {
    private final String name;
    private int nodesCounter;

    public NamedGraphNodeFactory() {
        this("Node");
    }

    public NamedGraphNodeFactory(@NotNull String name) {
        this.name = name;
        nodesCounter = 0;
    }

    @Override
    public GraphNode createNode() {
        return new NamedGraphNode(name + nodesCounter++);
    }
}

File: src/main/java/org/blacksun/graph/node/NamedGraphNodeFactory.kt
package org.blacksun.graph.node

import org.blacksun.graph.channel.ChannelFactory
import org.blacksun.utils.Config

open class NamedGraphNodeFactory @JvmOverloads constructor(protected val name: String = "Node", protected val factory: ChannelFactory = Config.config.getProperty("channelFactory")) : GraphNodeFactory {
    protected var nodesCounter: Int = 0

    init {
        nodesCounter = 0
    }

    override fun createNode(): GraphNode {
        return NamedGraphNode(name + nodesCounter++, factory)
    }
}
